File: Prolog/fatorial.pl
fatorial(0, R) :- R = 1, !.
fatorial(N, R) :- N1 is N - 1,
		 fatorial(N1, R1),
		 R is N * R1.


fat(1,1).
fat(X,F) :- X1 is X-1, fat(X1, F1), F is X * F1.

File: Swift/estrutura_de_repeticao/entrada_usuario/fatorial.swift
/*
Escrever um programa que calcule o fatorial de um número N, cujo valor é obtido através do usuário. 
*/


func fatorial(_ N: Int) -> Int {
  var resultado = 1
  for i in 1...N {
    resultado *= i
  }
  return resultado
}


print("Digite o valor de N: ", terminator: "")
let N = Int(readLine()!)!

print("O fatorial de \(N): \(fatorial(N)) ")

